### PR TITLE
Uplink rebalance

### PR DESCRIPTION
**Pull request by ghost**
- Airlock charges now cost 2 telecrystals (from 5)
- Airlock charges now detonate with the proper force
- A progress bar has been added when removing the charge from an airlock
- There is no more RNG chance for the charge to blow up even if you extract it successfully
- When charges blow up, they do 20 guaranteed brute damage to everyone nearby as the shrapnel from the airlock goes everywhere
- Many uplink item descriptions have been changed to better reflect their use
- Adds a new traitor item, the Dart Pistol. Bought for 4 telecrystals, it functions as a pocket-sized syringe gun that is quieter to fire.
- Boxes of riot foam darts have been tweaked; it costs 8 telecrystals less (seriously? 10??) and is not restricted to nuke ops.
- Toy Stetchkin pistols now cost 3 telecrystals, down from 6
- Syndicate cyborg energy swords now consume 50 energy per use, down from 500. They consumed way too much to be useful.
- Syndicate cyborgs now have playstyle explanations sent to them upon spawning

Syndicate medical cyborgs.
- A full surgery toolset (minus the saw, explained later)
- Defibrillator paddles, safeties off and can penetrate armor
- A Syndicate cyborg hypospray that quickly synthesizes reagents and dispenses the following: Restorative Nanites (heal most damage types quite quickly), Potassium Iodide, and Morphine
- A flash
- A CO2 jetpack
- A crowbar
- An operative pinpointer
- An energy saw. It's a circular saw by default, but can be activated to have as much power as an energy sword at the cost of consuming power on attack.

Choice between assault and medical cyborgs is offered upon using the teleporter. In addition, they have names to reflect their status and the Syndicate medical cyborg's old sprite has been touched up since it's actually getting use.

### CODE REVIEW

**Comment by bgobandit** [comment]
:+1:

**Comment by Jalleo** [comment]
Surplus price not changed for the crates though.

**Comment by Xhuis** [comment]
The surplus isn't a cost, it's the percentage of it being included.

**Comment by Xhuis** [comment]
I'm changing to a generalized uplink rebalance. Please tag this as DNM as I have more changes I plan to make.

**Comment by Xhuis** [comment]
I added medical syndiborgs. Currently they're not implemented, but take a look at the code and see how you like it.

**Comment by Niknakflak** [comment]
Putting old sprites to use :+1:

**Comment by SeikoTheSeal** [comment]
Really like the charge changes. It's a fun item to use as a traitor, but it costing 5tc, 1/4 of your total  TC, really made it not worth it.

:+1: On the Dart Pistol, something i wanted to see in the uplink for ages.

**Comment by Xhuis** [comment]
Syndicate mediborgs have been fully added. Their modules consist of:
* A full surgery toolset (minus the saw, explained later)
* Defibrillator paddles, safeties off and can penetrate armor
* A Syndicate cyborg hypospray that quickly synthesizes reagents and dispenses the following: Tricordrazine, Salbutamol, Mannitol, Mutadone, Antitoxin, and Potassium Iodide
* A flash
* A CO2 jetpack
* A crowbar
* An operative pinpointer
* An energy saw. It's a circular saw by default, but can be activated to have as much power as an energy sword at the cost of consuming more power while held.

File: code/modules/surgery/tools.dm [hunk 1]
```diff
@@ -86,6 +86,46 @@
 	origin_tech = "materials=1;biotech=1"
 	attack_verb = list("attacked", "slashed", "sawed", "cut")
 
+/obj/item/weapon/circular_saw/energy //For use by medical syndiborgs
```
Comment (RemieRichards) [inline code/modules/surgery/tools.dm]: this should be a subtype of the cyborg esword.
it has no real reason to be a circular saw (any item can be made to work in surgery), but it DOES have merits being an esword, the fact you've pasta'd it's code here.

Comment (Xhuis) [inline code/modules/surgery/tools.dm]: Can do. Any idea how to make it function as a circular saw for surgery purposes?

**Comment by Xhuis** [comment]
Squashed commits, this can be considered merge-ready.

**Comment by PKPenguin321** [comment]
pretty cool :+1:

File: code/modules/projectiles/guns/projectile.dm [hunk 2]
```diff
@@ -89,7 +89,7 @@
 
 /obj/item/weapon/gun/projectile/attack_hand(mob/user)
 	if(loc == user)
-		if(suppressed)
+		if(suppressed && !istype(src, /obj/item/weapon/gun/syringe/syndicate)) //Dart pistols can't be unsuppressed
```
Comment (RemieRichards) [inline code/modules/projectiles/guns/projectile.dm]: No, snowflake as heck.
Add a var. ```can_unsuppress```

**Comment by RemieRichards** [comment]
Sort that out and ping me.

**Comment by Xhuis** [comment]
@RemieRichards Conflicts and snowflake code are fixed

**Comment by Xhuis** [comment]
Changelog is added. This is merge-ready.

File: code/game/gamemodes/antag_spawner.dm [hunk 3]
```diff
@@ -119,16 +119,21 @@
...
 /obj/item/weapon/antag_spawner/borg_tele
-	name = "Syndicate Cyborg Teleporter"
-	desc = "A single-use teleporter used to deploy a Syndicate Cyborg on the field."
+	name = "syndicate cyborg teleporter"
+	desc = "A single-use teleporter used to deploy a Syndicate cyborg on the field."
 	icon = 'icons/obj/device.dmi'
 	icon_state = "locator"
 	var/TC_cost = 0
+	var/borg_to_spawn
+	var/list/possible_types = list("assault", "medical")
```
Comment (Miauw62) [inline code/game/gamemodes/antag_spawner.dm]: mind making this an associative list? it's slightly cleaner than using a switch()

File: code/game/machinery/doors/airlock.dm [hunk 4]
```diff
@@ -324,7 +324,9 @@ About the new airlock wires panel:
 /obj/machinery/door/airlock/examine(mob/user)
 	..()
 	if(charge && !p_open && in_range(user, src))
-		user << "The maintenance panel is bulging slightly."
+		user << "<span class='warning'>The maintenance panel is slightly bulging.</span>"
```
Comment (Miauw62) [inline code/game/machinery/doors/airlock.dm]: this isnt really your job to fix but the idea of spacemans shoving the maintenance panel on with so much force that they somehow /bend the airlock metal to make it bulge/ is p hilarious

File: code/modules/reagents/reagent_containers/borghydro.dm [hunk 5]
```diff
@@ -125,6 +125,15 @@ Borg Hypospray
...
 	if(empty)
 		usr << "<span class='warning'>It is currently empty! Allow some time for the internal syntheszier to produce more.</span>"
+
+/obj/item/weapon/reagent_containers/borghypo/syndicate
+	name = "syndicate cyborg hypospray"
+	desc = "A modified hypospray that generates reagents more quickly and consumes less power."
+	icon_state = "borghypo_s"
+	charge_cost = 20
+	recharge_time = 2
+	reagent_ids = list("tricordrazine", "salbutamol", "mannitol", "mutadone", "antitoxin", "potass_iodide")
```
Comment (Miauw62) [inline code/modules/reagents/reagent_containers/borghydro.dm]: you could add morphine to this or something to give them more combat capability? not sure how balanced that would be

File: code/game/machinery/doors/airlock.dm (hunk 4 above)
Comment (Xhuis) [inline code/game/machinery/doors/airlock.dm]: the fluff behind it is that the charge is bulky and they don't -completely- fasten the panel, just kinda slap it on

**Comment by Xhuis** [comment]
Syndicate cyborgs now have a string explaining their playstyle. Syndicate mediborgs' have hyposprays now have Restorative Nanites (all-purpose drug that heals most damage types), Potassium Iodide, and Morphine.

Merge-ready!

**Comment by Xhuis** [comment]
Conflicts removed, again.

**Pull request merged**